File: LeetCode/1-twosum.cpp
//求两数和
//Method-1，暴力枚举
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
    for (int i = 0; i < nums.size(); i++)
    {
        for (int j = i + 1; j < nums.size(); j++)
        {
            if (nums[i] + nums[j] == target)
            {
                return {i, j};
            }
        }
    }
    return {};
    }
};


//Method-2，哈希表

File: 力扣HOT-100/1-twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
      #利用哈希表
        hashtable = dict()
        for i, num in enumerate(nums):
            if target - num in hashtable:
                return [hashtable[target - num], i]
            hashtable[nums[i]] = i
        return []
